File: solutions/1431. Kids With the Greatest Number of Candies/1431.js
// Time complexity: O(n)
// Space complexity: O(n)
var kidsWithCandies = function(candies, extraCandies) {
    let ans = [];
    let maxCandies = 0;
    for (let candy of candies) {
        maxCandies = Math.max(maxCandies, candy);
    }

    for (let candy of candies) {
        ans.push(candy + extraCandies >= maxCandies);
    }
    return ans;
}

File: solutions/1431. Kids With the Greatest Number of Candies/1431.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
 public:
  vector<bool> kidsWithCandies(const vector<int>& candies, int extraCandies) {
    vector<bool> ans;
    int maxCandies = 0;

    for (const int candy : candies)
      maxCandies = max(maxCandies, candy);

    for (const int candy : candies)
      ans.push_back(candy + extraCandies >= maxCandies);

    return ans;
  }
};